test/runtime/atomic_test.cpp

```diff
@@ -0,0 +1,40 @@
+// The MIT License (MIT)
+//
+// Copyright (c) 2024 Nick Thompson
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+#include <catch2/catch_all.hpp>
+#include <mp-units/math.h>
+#include <mp-units/ostream.h>
+#include <mp-units/systems/angular/angular.h>
+#include <mp-units/systems/isq/space_and_time.h>
+#include <mp-units/systems/si/unit_symbols.h>
+#include <mp-units/systems/si/units.h>
+#include <atomic>
+
+using namespace mp_units;
+using namespace mp_units::si::unit_symbols;
+
+TEST_CASE("std::atomic works with dimensioned types", "[atomic][assignment]")
+{
+  std::atomic<isq::area[m2]> a = 3.0 * isq::area[m2];
+  std::atomic<isq::area[m2]> b = 3.0 * isq::area[m2];
```

Comment:
Did you mean?

```suggestion
  std::atomic<quantity<isq::area[m2]>> a = 3.0 * isq::area[m2];
  std::atomic<quantity<isq::area[m2]>> b = 3.0 * isq::area[m2];
```

With the above everything seems to work fine: https://godbolt.org/z/zqGq18r3f.